### PR TITLE
Header and navigation

### PR DESCRIPTION
**Pull request by boxadesign**
### What is the context of this PR?

Added the new header design and header navigation components for internal and external variants.

Created horizontal section/sub page navigation component. 

### How to review 
Check the code against design. Check on mobile and for print, no js and accessibility.

The header and navigation have already been reviewed in separate PR's (links below). There were significant changes/improvments made in this branch to those components so have rolled it into this PR.

Header - https://github.com/ONSdigital/sdc-global-design-patterns/pull/111

Nav - https://github.com/ONSdigital/sdc-global-design-patterns/pull/107

#### External header and nav:

image

#### Internal header and nav:

image

#### Horizontal section navigation:

image

### CODE REVIEW

File: components/button/_buttons.scss [hunk 1]
```diff
@@ -158,3 +158,67 @@
...
+  height: 28px;
+  text-indent: -9999px;
+  background: transparent;
+  border: transparent;
+  border-radius: 0;
+  &:focus, &:hover {
+    background-color: transparent;
+    border-color: transparent;
+  }
+  span {
```
Comment (saedwards) [inline components/button/_buttons.scss]: Think we need to follow the convention of using BEM in the pattern library, entity selectors are discouraged and would be better to use an element selector inside this modifier.

File: components/button/button--header.hbs [hunk 2]
```diff
@@ -1,3 +1,3 @@
-<div class="bar u-pl-m u-pt-m">
+<div class="bar u-pl-m u-pt-m u-pb-m">
```
Comment (saedwards) [inline components/button/button--header.hbs]: Potential overuse of utility classes, should be used sparingly. Prefer adding these single properties into the block, or create a modifier if a new variant is needed.

File: components/button/_buttons.scss (hunk 1 above)
Comment (boxadesign) [reply components/button/_buttons.scss]: Yep good point.

File: components/button/button--header.hbs (hunk 2 above)
Comment (boxadesign) [reply components/button/button--header.hbs]: As discussed, these classes are only added to display the button and isn't part of the code that would be used.

**Comment by boxadesign** [comment]
@saedwards @armstrongb Have added a change based on comments.

File: components/04-navigation/section-navigation/horizontal-navigation/horizontal-navigation.hbs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<nav class="{{nav-wrapper-classes}}" role="menu">
```
[the remark this answers is not in the document]
Comment (boxadesign) [reply components/04-navigation/section-navigation/horizontal-navigation/horizontal-navigation.hbs]: My reason for using `menu` is that as we are using the `nav` element we don't need to add a `role` (http://adrianroselli.com/2017/10/dont-use-aria-menu-roles-for-site-nav.html). 

But as we are using a menu button for mobile I included the `role=menu` as shown here: Have a look at this https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html

**Comment by boxadesign** [comment]
@saedwards I've made the changes you requested. Can this now be approved?

**Pull request merged**